lib/bugsnag.ex

```diff
@@ -73,14 +74,22 @@ defmodule Bugsnag do
   end
 
   defp send_notification(body) do
-    HTTPoison.post(@notify_url, body, @request_headers)
+    release_stage = Application.get_env(:bugsnag, :release_stage)
+    notify_release_stages = Application.get_env(:bugsnag, :notify_release_stages)
+
+    if (release_stage && length(notify_release_stages) && Enum.member?(notify_release_stages, release_stage)) do
+      HTTPoison.post(@notify_url, body, @request_headers)
+    else
+      {:ok, %{reason: 'not_sent'}}
+    end
   end
 
   defp default_config do
     [
       api_key:       {:system, "BUGSNAG_API_KEY", "FAKEKEY"},
       use_logger:    {:system, "BUGSNAG_USE_LOGGER", true},
-      release_stage: {:system, "BUGSNAG_RELEASE_STAGE", "test"}
+      release_stage: {:system, "BUGSNAG_RELEASE_STAGE", "test"},
```

Comment:
This conflicts with code in `Bugsnag.Payload` which sets the release stage to "production" by default and might cause some confusion. Was there a particular reason for defaulting to test?

I was thinking that maybe compiling in the mix env would make a good default? That *should* keep most people happy.